14-tidyr.Rmd

```diff
@@ -0,0 +1,154 @@
+
+However, converting to the pure 'long' format is the easiest way to convert this data to the pure 'wide' format. Manually entering the data into a spreadsheet is more intuitive in the wide format and some plotting functions actually work better in the wide format.
+
+Before we really start, let's re-organize the ID and Observations variables so that they are on order (this will make more sense later). We're swapping `pop` and `continent`
+
+
+```{r}
+gapminder <- gapminder[,c(1,2,4,3,5,6)]
+str(gapminder)
+
+```
+
+
+#### From intermediate to long format with gather()
+The `tidyr` function `gather()` will 'gather' your observation variables into a single variable.
+
+```{r}
+gap_long <- gapminder %>% gather(obs_type,obs_values,pop,lifeExp,gdpPercap)
+str(gap_long)
+```
+
+Here we have used piping syntax which is similar to what we were doing in the previous lesson with dplyr. In fact, these are compatible and you can use a mix of tidyr and dplyr functions by piping them together
+
+Inside `gather()` we first name the new column for the new ID variable (`obs_type`), the name for the new amalgamated observation variable (`obs_value`), then the names of the old observation variable (`pop`,`lifeExp`,`gdpPercap`). Gather also allows the alternative syntax of using the `-` symbol to identify which variables are not to be gathered (i.e. ID variables)
+
+```{r}
+gap_long <- gapminder %>% gather(obs_type,obs_values,-continent,-country,-year)
+str(gap_long)
+```
+
+That may seem trivial with this particular dataframe, but sometimes you have 1 ID variable and 40 Observation variables. The flexibility is a huge time saver!
+
+> #### Challenge 1 {.challenge}
+>
+> Using `gap_long`, can you calculate the mean life expectancy, population, and gdpPercap for each continent?
+>**Hint:** use the `group_by()` and `summarize()` functions we learned in the dplyr lesson
+>
+
+#### From long to wide format with spread()
+Now just to double-check our work, lets use the opposite of `gather()` to spread our observation variables back out with the aptly named `spread()`
+
+```{r}
+gap_normal <- gap_long %>% spread(obs_type,obs_values)
+identical(gap_normal,gapminder)
+```
+
+That's great! We've gone from the longest format back to the intermediate and we didn't introduce any errors in our code.
+
+Now lets convert the long all the way to the wide. In the wide format, we will keep country and continent as ID variables and spread the observations across the 3 metrics (`pop`,`lifeExp`,`gdpPercap`) and time (`year`). First we need to create appropriate labels for all our new variables (time*metric combinations) and we also need to unify our ID variables to simplify the process of defining `gap_wide`
```

Comment:
I think this explanation is still a bit complicated. Maybe simplify a bit further?